clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -743,76 +743,76 @@ runSYCLPostLinkTool(ArrayRef<StringRef> InputFiles, const ArgList &Args) {
     if (!ImageFileOrErr)
       return ImageFileOrErr.takeError();
 
-    std::vector Modules = {module_split::SplitModule(
+    std::vector Modules = {module_split::ProcessedModule(
         *ImageFileOrErr, util::PropertySetRegistry(), "")};
     return Modules;
   }
 
-  return llvm::module_split::parseSplitModulesFromFile(*TempFileOrErr);
+  return llvm::module_split::parseProcessedModulesFromFile(*TempFileOrErr);
 }
 
-/// Invokes SYCL Split library for SYCL offloading.
+/// Invokes SYCL Post Link library for SYCL offloading.
```

Comment:
I don't think "Post Link" is a good choice of wording here. This name reflects some requirements for using this library in compilation process, but I think they are false in general. 
For example, in `no-rdc` mode, there is no "link" step as the code is self-contained by definition.

I think something like "SYCL offload finalization library" better reflects the purpose of this code.

Same applies to `SYCLPostLinkProcess` function from llvm/include/llvm/SYCLLowerIR/ModuleSplitter.h.

Reply:
`SYCLOffloadFinalize` ?

Reply:
In fact, in `no-rdc` mode this processing can be done in `-c` phase because they are already implemented as transformation passes. In that case, there is no need to invoke this library at all in `clang-linker-wrapper`.

Reply:
Friendly ping, @bader

Reply:
Friendly pong.

> SYCLOffloadFinalize ?

Could you clarify what are you trying to ask here?

Reply:
Do you think that `SYCLPostLinkProcess` is better to be replaced by `SYCLOffloadFinalize`?

What do you think about my comment regarding `no-rdc`?

Reply:
> Do you think that `SYCLPostLinkProcess` is better to be replaced by `SYCLOffloadFinalize`?

`SYCLOffloadFinalize` doesn't seem to fit perfectly, but it sounds better than `SYCLPostLinkProcess` in my opinion.

> What do you think about my comment regarding `no-rdc`?

I agree, but I don't get the motivation to have a separate "library". Correct me if I wrong, but all this library does is running LLVM passes. My understanding is that we can integrate these LLVM passes either into standard optimization pipeline (preferred) or by building a separate optimization pipeline. @sarnex is using this kind of approach in https://github.com/intel/llvm/pull/15083/files#diff-f7dc51acede945c7c8481147e773e9c09230a93b937bee9eab08ee52826cfa8eR1779.

Basically, I would consider integrating new passes into LLVM optimization pipeline for both `rdc` and `no-rdc` modes.

Reply:
Module splitting can't be integrated into `-c` phase even if it were be implemented as a Pass.

Eventually, I see the compilation pipeline as thin-LTO + optimization pipeline. But I don't see that it happens soon. The plan is to move `sycl-post-link` functionality in `clang-linker-wrapper`. Then move to the thin-LTO once it is ready.